hbase-server/src/main/java/org/apache/hadoop/hbase/io/util/MemorySizeUtil.java

```diff
@@ -216,6 +216,9 @@ public static float getBlockCacheHeapPercent(final Configuration conf) {
   public static long getOnHeapCacheSize(final Configuration conf) {
     float cachePercentage = conf.getFloat(HConstants.HFILE_BLOCK_CACHE_SIZE_KEY,
       HConstants.HFILE_BLOCK_CACHE_SIZE_DEFAULT);
+    // only uses when it's ClientSideRegionScanner or user explicitly sets it
+    long fixedCacheSize = conf.getLong(HConstants.HBASE_BLOCK_CACHE_FIXED_SIZE_KEY,
```

Comment:
So here, we are exposing this fixed size for L1 cache as a generic thing.  At RS level also some one can use this. Well that is fine.  On large heap sized RS, this make sense also.  But may be we have to do more checks I fear?
Like in current way of both L1 size and memstore size as % of Xmx, we have checks to make sure addition of wont exceed some 80% of Xmx..  Now when one is used as fixed size and other as %, how we can make sure such things?  Or else a wrong configured RS can cause OOME!  We can add this later in a follow up Jira also.  But if we are exposing this as a global thing, lets be sure of doing all safety checks.  Or else lets make this usable only at that special client scanner level